.cdsrc.json

```diff
@@ -1,21 +1,32 @@
 {
-    "build": {
-        "target": ".",
-        "tasks": [
-            {
-                "for": "java"
-            },
-            {
-                "for": "mtx",
-                "src": ".",
-                "dest": "mtx-sidecar"
-            },
-            {
-                "for": "hana"
-            }
-        ]
+  "build": {
+    "target": ".",
+    "tasks": [
+      {
+        "for": "java"
+      },
+      {
+        "for": "mtx",
+        "src": ".",
+        "dest": "mtx-sidecar"
+      },
+      {
+        "for": "hana"
+      }
+    ]
+  },
+  "hana": {
+    "deploy-format": "hdbtable"
+  },
+  "requires": {
```

Comment:
`requires` section in `.cdsrc.json` should not be relevant for Java applications. Why is this added?

Reply:
This is required by `cds-dk`. For now, it cannot read the project's features from `application.yaml`.

Reply:
Would it also be possible to pass these information on the command line?
I don't really like to fill the `.cdsrc.json` with information that is just required for CLI calls like `cds add helm`.

Reply:
That information isn't just required by `cds add helm`. It is required by the `cds-dk`. You can execute `cds add mta` and see that the configuration added by it is incorrect as well. Yes we can execute `cds add hana` or `cds add xsuaa` to get the correct configuration added for the deployment but these commands corrupt the files in the project. For example:

- `cds add hana` changes the `build` script in `package.json` of `db`.
- `cds add multitenancy` corrupts the `xs-security.json` file.

BTW, now that `streamlined-mtx` branch is merged and `.cdsrc.json`, I see that it is set up for multitenancy only. I think we will need two files here. One for Single Tenant and one for Multi Tenant (similar to mta.yaml files). 

Will it be okay if I create the two files (`.cdsrc-st.json` and `.cdsrc-mt.json`) and place them in a folder `kyma (?)`.  In the instructions, I will then mention that user needs to replace the contents of `.cdsrc.json` in root folder with the contents of the file in `kyma` folder.

Reply:
> You can execute cds add mta and see that the configuration added by it is incorrect as well.

I guess that is why we added predefined mta.yaml files..

> BTW, now that streamlined-mtx branch is merged and .cdsrc.json, I see that it is set up for multitenancy only

No, at the moment you can still deploy the app as single-tenant, even with the same `.cdsrc.json`. I just verified that again. Why is there a difference required in case of Kyma?

Reply:
When `cds add helm` is executed, `cds-dk` reads the project's configuration and checks which all `features` the project has. After this, it adds the required configuration for each `feature`. 

`cds-dk` uses various sources to check if the project has a specific `feature` or not. One of the source is `.cdsrc.json` file. Now that `.cdsrc.json` file has this specific key `requires.multitenancy` set to `true`, `cds-dk` will think that the project is MultiTenant. So every time `cds add helm` is executed, `cds-dk` will add the multitenant configuration only.

This is why I suggest having two `.cdsrc.json` files.

Reply:
Hm, I don't really like adding two different `.cdsrc.json` files just for `cds add helm`. Is it possible to specify those `features` on the command line as well, when calling `cds add helm`. This would make us independent of the specific information stored in `.cdsrc.json` or in other places.

Reply:
not really. The only other option I see is adding pre defined charts for single tenant and multi tenant just like `mta.yaml`.

Reply:
> The only other option I see is adding pre defined charts for single tenant and multi tenant just like mta.yaml.

That is probably a better option then. However I think this really makes sense with the charts provided by the Unified Runtime, as those are easily "updatable". @davidhunglam made first tests with it and the bookshop in https://github.com/SAP-samples/cloud-cap-samples-java/pull/197. Maybe it makes sense to discuss these topics together in a meeting :)